Various edits from the roadmap. (#109)
Heegu-sama/Homm3BG: sections/combat.tex

}

\subsection*{\hypertarget{Combatterminology}{Combat Terminology}}
The following terms are used to describe effects and elements during Combat:\par

> \textbf{Attacking Playe}r – The player who started the Combat.\par

Reviewer: ```suggestion
\textbf{Attacking Player} – The player who started the Combat.\par
```

You may only use \textbf{one Spell per Combat Round}.
Other card types can be used as many times as you want to.
Ongoing \includesvg[height=10px]{\svgs/ongoing.svg} and \includesvg[height=10px]{\svgs/activation.svg} Activate effects can be used only \textbf{when Activating one of your Units and before it attacks}.
Ongoing effects\index{Ongoing Effects} last until end of Combat or if the effect on the card is used up.\par
Instant \includesvg[height=10px]{\svgs/instant.svg} Cards may be played \textbf{at any time} except between rolling the Combat Die and resolving damage unless otherwise stated.

> Cards which increase a Unit's \includesvg[height=10px]{\svgs/attack.svg} or \includesvg[height=10px]{\svgs/defense.svg}, such as the Attack and Defense Statistic cards, \textbf{must} be played after an attack has been announced and persist only for the duration of that attack.

Reviewer: Why no "instant"? Isn't it a term from the original?

If either the Difficulty of the Neutral Field or the Level of a defeated enemy Main Hero was equal to your Level, gain 1 \includesvg[height=10px]{\svgs/exp.svg}.
If they were higher than your Level, gain 2 \includesvg[height=10px]{\svgs/exp.svg}.
Winning a Neutral Combat against a Neutral Azure \includesvg[height=10px]{\svgs/azure.svg} Unit grants your Hero Level 7 immediately.
If you ever gain multiple Levels at the same time, resolve their effects in order.
Level ups must be resolved before Visiting the Field where the Combat happened.\par
Secondary Heroes cannot ever gain Experience.

> You also \textbf{do not gain} Experience from defeating a Secondary Hero, or if an enemy Hero Surrenders to you.\par

Reviewer: ```suggestion
You also \textbf{do not gain} Experience from defeating a Secondary Hero, or if an enemy Hero Surrenders to you.
```

Are those necessary? I guess `\subsection` implies a paragraph.

The Spell Deck should be used when instructed by the cards in the AI Deck.
If an AI Hero is instructed to draw a card, they will draw and resolve another card from the AI Deck.\par
During Combat against Neutral enemies or AI Heroes, their Units follow an automatic set of instructions:
\par
\textbf{Initiative rules} work identically.
\includesvg[height=10px]{\svgs/unit_ground.svg} and \includesvg[height=10px]{\svgs/unit_flying.svg} Units prioritize attacking Units of the same tier.
If this is impossible, they attack the closest Unit, prioritizing the lowest tier one.
\includesvg[height=10px]{\svgs/unit_ranged.svg} Units prioritize attacking other \includesvg[height=10px]{\svgs/unit_ranged.svg} Units of the same tier, then lower tier, and finally higher tier.
If there are no \includesvg[height=10px]{\svgs/unit_ranged.svg} Units for them to target they prioritize \includesvg[height=10px]{\svgs/unit_ground.svg} and \includesvg[height=10px]{\svgs/unit_flying.svg} Units in the same tier order.
If there's more than one valid target, they attack the closest one.\par

> If there's ever a tie between equally valid targets for any Units, the players choose which Unit is attacked. Enemy units cannot \hyperlink{Defend}{Defend} unless instructed to.

Reviewer: ```suggestion
If there's ever a tie between equally valid targets for any Units, the player chooses which Unit is attacked. Enemy units cannot \hyperlink{Defend}{Defend} unless instructed to.
```
Well, since those rule apply for a solo game

> \textbf{AI Heroes} always start in their Town's Field unless otherwise stated.

Reviewer: I might be overthinking this, but here's how I see it. Maybe half of this subsection is about actual combat. There could be a separate chapter called "AI Heroes" with this text here starting it. "Player vs AI" subsection would be called "Player vs Neutral Units", which would describe combat only (_During Combat against Neutral enemies [...] Enemy Units cannot Defend unless instructed to._)
The rest of the text (about cards, game modes) would go to AI Heroes, with a hyperlink to "Player vs Neutral Units", explaining that AI Heroes follow the same logic while fighting. WDYT?

Author: I'll make a separate PR for this later. It's clear that some re-organization should be done about these.
